dev_environment/setup_linux_dev_env.sh

```diff
@@ -0,0 +1,187 @@
+apt-get install -y bc \
+      bison \
+      build-essential \
+      clang \
+      curl \
+      exuberant-ctags \
+      flex \
+      gcc-8 \
+      gnutls-bin \
+      grafana \
+      jq \
+      libc6-dev \
+      libcurl4-openssl-dev \
+      libelf-dev \
+      libjson-c-dev \
+      libncurses5-dev \
+      libpcap-dev \
+      libssl-dev \
+      linux-headers-generic \
+      linux-tools-common \
+      linux-tools-generic \
+      llvm \
+      prometheus \
+      rsync
+
+#install the latest go lang version
+  os=`uname|tr '[:upper:]' '[:lower:]'`
+  go_filename=`curl -s https://go.dev/dl/?mode=json|jq '.[0].files[].filename'|grep $os|grep $arch|egrep -v "ppc"|tr -d '"'`
+  wget https://go.dev/dl/$go_filename
+  tar -C /usr/local -xzf $go_filename && rm -f $go_filename
+  echo export PATH=$PATH:/usr/local/go/bin >> /root/.bashrc
+
+# clone the l3afd repo in to root directly
+# can use mapped directory i.e. at /home/ubuntu/Home
+if [ -d "/root/l3afd" ];
+then
+  echo "repo already exited"
+else
+  git clone https://github.com/l3af-project/l3afd.git
+  git clone https://github.com/l3af-project/l3af-arch.git
+fi
+
+# Copy grafana configs and dashboards into place
+if [ -d "/var/lib/grafana/dashboards" ];
+then
+  echo "grafana directory already existed"
+else
+  mkdir -p /var/lib/grafana/dashboards
+fi
+chown grafana:grafana /var/lib/grafana/dashboards
+cp /root/l3af-arch/dev_environment/cfg/grafana/dashboards/*.json /var/lib/grafana/dashboards
+chown grafana:grafana /var/lib/grafana/dashboards/*.json
+cp /root/l3af-arch/dev_environment/cfg/grafana/provisioning/dashboards/l3af.yaml /etc/grafana/provisioning/dashboards
+chown root:grafana /etc/grafana/provisioning/dashboards/*.yaml
+cp /root/l3af-arch/dev_environment/cfg/grafana/provisioning/datasources/l3af.yaml /etc/grafana/provisioning/datasources
+chown root:grafana /etc/grafana/provisioning/datasources/*.yaml
+
+# Copy prometheus config and restart prometheus
+cp /root/l3af-arch/dev_environment/cfg/prometheus.yml /etc/prometheus/prometheus.yml
+if uname -a | grep -q 'WSL';
+then
+  echo "WSL DETECTED"
+  apt-get install daemon
+  #start/restart prometheus-node-exporter
+  /etc/init.d/prometheus-node-exporter start
+  /etc/init.d/prometheus-node-exporter stop
+  /etc/init.d/prometheus-node-exporter start
+
+  # Start and enable Grafana
+  sleep 1
+  /etc/init.d/grafana-server stop || true
+  /etc/init.d/grafana-server stop || true
+  /etc/init.d/grafana-server start || true
+else
+  # the configuration got copied, restart the prometheus service
+  systemctl daemon-reload
+  systemctl restart prometheus prometheus-node-exporter
+
+  # Start and enable Grafana
+  systemctl start grafana-server
+  systemctl enable grafana-server.service
+fi
+
+# Get Linux source code to build our eBPF programs against
+if [ ! -d "/usr/src/linux" ];
+then
+  git clone --branch v5.1 --depth 1 https://github.com/torvalds/linux.git /usr/src/linux
+fi
```

Comment:
if dir exists print message that this was skipped

Reply:
Add the skip message